scapy/layers/can.py

```diff
@@ -84,6 +88,211 @@ def extract_padding(self, p):
+            min_val = -2**(self.size - 1)
+            max_val = 2**(self.size - 1) - 1
+        else:
+            min_val = 0
+            max_val = 2 ** self.size - 1
+
+        min_val = round(min_val * self.scaling + self.offset, self.ndigits)
+        max_val = round(max_val * self.scaling + self.offset, self.ndigits)
+
+        return RandFloat(min(min_val, max_val), max(min_val, max_val))
+
+    def i2len(self, pkt, x):
+        return float(self.size) / 8
+
+
+class LEUnsignedSignalField(SignalField):
+    def __init__(self, name, default, start, size, scaling=1, unit="",
+                 offset=0, ndigits=3):
+        SignalField.__init__(self, name, default, start, size,
+                             scaling, unit, offset, ndigits, "<B")
+
+
+class LESignedSignalField(SignalField):
+    def __init__(self, name, default, start, size, scaling=1, unit="",
+                 offset=0, ndigits=3):
+        SignalField.__init__(self, name, default, start, size,
+                             scaling, unit, offset, ndigits, "<b")
+
+
+class BEUnsignedSignalField(SignalField):
+    def __init__(self, name, default, start, size, scaling=1, unit="",
+                 offset=0, ndigits=3):
+        SignalField.__init__(self, name, default, start, size,
+                             scaling, unit, offset, ndigits, ">B")
+
+
+class BESignedSignalField(SignalField):
+    def __init__(self, name, default, start, size, scaling=1, unit="",
+                 offset=0, ndigits=3):
+        SignalField.__init__(self, name, default, start, size,
+                             scaling, unit, offset, ndigits, ">b")
+
+
+class LEFloatSignalField(SignalField):
+    def __init__(self, name, default, start, size, scaling=1, unit="",
+                 offset=0, ndigits=3):
+        SignalField.__init__(self, name, default, start, size,
+                             scaling, unit, offset, ndigits, "<f")
+
+
+class BEFloatSignalField(SignalField):
+    def __init__(self, name, default, start, size, scaling=1, unit="",
+                 offset=0, ndigits=3):
+        SignalField.__init__(self, name, default, start, size,
+                             scaling, unit, offset, ndigits, ">f")
+
+
+class SignalPacket(Packet):
+    def pre_dissect(self, s):
+        if not all([isinstance(f, SignalField) for f in self.fields_desc]):
+            raise Scapy_Exception("Use only SignalFields in a SignalPacket")
+        return s
+
+    def post_dissect(self, s):
+        """ SignalFields can be dissected on packets with unordered fields.
+        The order of SignalFields is defined from the start parameter.
+        After a build, the consumed bytes of the length of all SignalFields
+        have to be removed from the SignalPacket.
+        """
+        if self.wirelen > 8:
+            raise Scapy_Exception("Only 64 bits for all SignalFields "
+                                  "are supported")
+        self.raw_packet_cache = None  # Reset packet to allow post_build
```

Comment:
Do you really need this ?

This is used in very few places in Scapy, when we rely on `post_build` to do some tricks. It's used a bit in some Low endian protocols, but mostly in `FCSField`. It's purpose is to force Scapy to call `post_build`, therefore ignore the cache.

It doesn't seem there's a `post_build` function, nor any need for this call :/

Reply:
Hi, all my dissect tests fail, without this line.